In [1]:
import numpy as np

1. Calculating the posistion and size of the image

In [23]:
lens1 = np.matrix([[1,0],[-1/52,1]])

In [24]:
gap1 = np.matrix([[1,15],[0,1]])

In [25]:
lens2 = np.matrix([[1,0],[-1/40,1]])

In [26]:
gap2 = np.matrix([[1,70-15],[0,1]])

In [27]:
lens3 = np.matrix([[1,0],[-1/19,1]])

In [28]:
gap3 = np.matrix([[1,90-70],[0,1]])

In [29]:
lens4 = np.matrix([[1,0],[-1/33,1]])

In [30]:
systemMatrix = lens4*gap3*lens3*gap2*lens2*gap1*lens1

In [31]:
print(systemMatrix)

[[-0.67067308  9.90131579]
 [ 0.05301573 -2.27372408]]


In [19]:
so = 56

In [32]:
si = -(so*systemMatrix[0,0]+systemMatrix[0,1])/(so*systemMatrix[1,0]+systemMatrix[1,1])

In [33]:
print(si)

39.78435819262387


In [37]:
mag1= systemMatrix[0,0]

In [38]:
print(mag1)

-0.6706730769230771


2. Determine whcih element is the aperature stop

In [101]:
matrix1 = np.matrix([[1,56],[0,1]])

In [119]:
alpha_o1 = abs(25/matrix1[0,1])

In [120]:
print(alpha_o1)

0.44642857142857145


In [104]:
matrix2 = gap1*lens1*matrix1

In [117]:
alpha_o2 = abs(20/matrix2[0,1])

In [118]:
print(alpha_o2)

0.36465638148667606


In [107]:
matrix3 = gap2*lens2*matrix2

In [115]:
alpha_o3 = abs(8.5/matrix3[0,1])

In [116]:
print(alpha_o3)

0.34276851492826665


In [110]:
matrix4 = gap3*lens3*matrix3

In [113]:
alpha_o4 = abs(22.5/matrix4[0,1])

In [114]:
print(alpha_o4)

0.8135556002854579


The aperture stop is Lens 3 (f=19 and d=17)

3. Find the locations of the entrance and exit pupils

so_ent = 55

In [124]:
entMatrix = lens1*gap1*lens2

In [125]:
si_ent =  -(so_ent*entMatrix[0,0]+entMatrix[0,1])/(so_ent*entMatrix[1,0]+entMatrix[1,1])

In [126]:
print(si_ent)

37.27767695099818


Entrance pupil at 37.278 to to the left of lens 1 or x=-37.278

In [128]:
so_ex = 20

In [129]:
exMatrix = lens4

In [130]:
si_ex = -(so_ex*exMatrix[0,0]+exMatrix[0,1])/(so_ex*exMatrix[1,0]+exMatrix[1,1])

In [131]:
print(si_ex)

-50.769230769230774


Exit pupil at x=39.231 (50.769 to the left of lens 3)

4. Find the field stop

In [132]:
d = 56-si_ent

In [133]:
print(d)

18.722323049001822


In [136]:
alpha_o11 = abs(25/(matrix1[0,1]-matrix1[0,0]*d))

In [137]:
print(alpha_o11)

0.6706426484907498


In [145]:
alpha_o21 = abs(20/(matrix2[0,1]-matrix2[0,0]*d))

In [146]:
print(alpha_o21)

0.4816433566433567


In [143]:
alpha_o31 = abs(8.5/(matrix3[0,1]-matrix3[0,0]*d))

In [144]:
print(alpha_o31)

2392537302040576.0


In [147]:
alpha_o41 = abs(22.5/(matrix4[0,1]-matrix4[0,0]*d))

In [148]:
print(alpha_o41)

1.4900841346153848


Field stop will be at lens 2.